Auth0/ClaimValidators.swift

```diff
@@ -261,10 +261,15 @@ struct IDTokenAuthTimeValidator: JWTValidator {
 
     func validate(_ jwt: JWT) -> LocalizedError? {
         guard let authTime = jwt.claim(name: "auth_time").date else { return ValidationError.missingAuthTime }
+
         let currentTimeEpoch = baseTime.timeIntervalSince1970
-        let authTimeEpoch = authTime.timeIntervalSince1970 + Double(maxAge) + Double(leeway)
-        guard authTimeEpoch < currentTimeEpoch else {
-            return ValidationError.pastLastAuth(baseTime: currentTimeEpoch, lastAuthTime: authTimeEpoch)
+        let authEpoch = authTime.timeIntervalSince1970
+
+        let authAge = currentTimeEpoch - authEpoch
+        let adjustedMaxAge = Double(maxAge) + Double(leeway)
+
+        guard authAge <= adjustedMaxAge else {
+            return ValidationError.pastLastAuth(baseTime: currentTimeEpoch, lastAuthTime: authEpoch)
```

Comment:
[The spec](https://openid.net/specs/openid-connect-core-1_0.html) states:

> Specifies the allowable elapsed time in seconds since the last time the End-User was actively authenticated by the OP. If the elapsed time is **greater** than this value, the OP MUST attempt to actively re-authenticate the End-User. (The max_age request parameter corresponds to the OpenID 2.0 PAPE [OpenID.PAPE] max_auth_age request parameter.) When max_age is used, the ID Token returned MUST include an auth_time Claim Value.